spatialdata/utils.py

```diff
@@ -19,15 +26,106 @@ def phase3(self) -> None:
         self.same_files, self.diff_files, self.funny_files = fcomp
 
 
-def are_directories_identical(dir1: Any, dir2: Any) -> bool:
+def are_directories_identical(
+    dir1: Any,
+    dir2: Any,
+    exclude_regexp: Optional[str] = None,
+    _root_dir1: Optional[str] = None,
+    _root_dir2: Optional[str] = None,
+) -> bool:
     """
     Compare two directory trees content.
     Return False if they differ, True is they are the same.
     """
+    if _root_dir1 is None:
+        _root_dir1 = dir1
+    if _root_dir2 is None:
+        _root_dir2 = dir2
+    if exclude_regexp is not None:
+        if re.match(rf"{_root_dir1}/" + exclude_regexp, str(dir1)) or re.match(
+            rf"{_root_dir2}/" + exclude_regexp, str(dir2)
+        ):
+            return True
+        ##
+        # loc = ZarrLocation(_root_dir1)
+        # reader = Reader(loc)()
+        # nodes = list(reader)
+        # assert len(nodes) == 1
+        # node = nodes[0]
+        # im_read = node.load(Multiscales).array(resolution="1", version=CurrentFormat().version)
+        #
+        # loc = ZarrLocation(_root_dir2)
+        # reader = Reader(loc)()
+        # nodes = list(reader)
+        # assert len(nodes) == 1
+        # node = nodes[0]
+        # im_read2 = node.load(Multiscales).array(resolution="1", version=CurrentFormat().version)
+        # print(im_read[:5, :5, 0].compute())
+        # print(im_read2[:5, :5, 0].compute())
+        ##
```

Comment:
I'm not sure hwat to do with this text, I would just remove otherwise not comment

Reply:
removed